upgrade_downgrade/test_canary_upgrade_downgrade.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -o pipefail
+set -x
+
+# TODO: Remove hardcoded versions and pass it from run_upgrade_downgrade_test
+# as command-line arguments like test_upgrade_downgrade.sh
+FROM_VERSION="istio-1.7-alpha.a8c04f92e236676977b3ed58132437d9fafc9aed"
+TO_VERSION="istio-1.8-alpha.f0f221ae1024fb2d4a93b355546f094376fb2e47"
+ISTIOPATH="$GOPATH/src/istio.io"
+
+FROM_DIR="${ISTIOPATH}/tools/upgrade_downgrade/from_dir.g9xSYb/${FROM_VERSION}"
+TO_DIR="${ISTIOPATH}/tools/upgrade_downgrade/to_dir.MrCCp4/${TO_VERSION}"
+TMP_DIR="${ISTIOPATH}/tools/upgrade_downgrade/templates"
+POD_FORTIO_LOG="${TMP_DIR}/fortio.log"
+
+FROM_ISTIOCTL="${FROM_DIR}/bin/istioctl"
+TO_ISTIOCTL="${TO_DIR}/bin/istioctl"
+
+# Maximum % of 503 response that cannot exceed
+MAX_503_PCT_FOR_PASS="15"
+# Maximum % of connection refused that cannot exceed
+# Set it to high value so it fails for explicit sidecar issues
+MAX_CONNECTION_ERR_FOR_PASS="30"
+SERVICE_UNAVAILABLE_CODE="503"
+CONNECTION_ERROR_CODE="-1"
+
+
+# Install Istio 1.7 minimal profile (only istiod)
+echo "Installing Istio ${FROM_VERSION}"
+${FROM_ISTIOCTL} install -y --set profile=minimal
+kubectl wait --all --for=condition=Ready pods -n istio-system --timeout=5m
+
+# Deploy sample application
+kubectl create namespace test
+kubectl label namespace test istio-injection=enabled
+
+echo "Deploying Echo v1 and v2"
+kubectl apply -f "${TMP_DIR}/fortio.yaml" -n test
+kubectl wait --all --for=condition=Ready pods -n test --timeout=5m
+
+echo "Generate internal traffic for echo v1 and v2"
+kubectl apply -f "${TMP_DIR}/fortio-cli.yaml" -n test
+
+# Install Istio 1.8 minimal profile with canary revision
+echo "Installng Istio-canary ${TO_VERSION}"
+${TO_ISTIOCTL} install -y --set profile=minimal --set revision=canary
```

Comment:
let's use targetver or something as a a label - we want to move away from using canary as tag because it confuses users

Reply:
Done. Had to change a bit because it should be a valid DNS name as it is appended to `istiod`.
```
TO_REVISION=$(echo "${TO_TAG}" | tr '.' '-')
```